### DISCUSSION TITLE
constant noise in analog output

### DISCUSSION THREAD

**Discussion by namboodirilab** [pyPhotometry/code#15] [Q&A]
Hi there,
We've been using pyphotometry systems for a while, and recently one of them started to have this weird ~3hz constant noise in analog output. The patch cord was not connected to anything, and it even happens when the led power was lowered to zero (second photo). Does anyone have idea what would be causing this? 

![noise](https://github.com/pyPhotometry/code/assets/91919007/67e3e271-4d9d-44e7-bcee-0a22d70b85a1)
![noise_zeroled](https://github.com/pyPhotometry/code/assets/91919007/7496f88a-9333-4b5a-ac74-c0095b3be7d9)

Thank you!

Best,
Huijeong

**Comment by ThomasAkam** [comment]
Hi Huijeong,

Could you try disconnecting the BNC cables from the photorecievers to the pyPhotometry board and taking a screenshot of what the signals look like in the continous acquisition mode.  This will allow us to see if the oscillations are being generated in the pyPhotometry board itself or in the photorecievers.

Could you also let me know what optical hardware and photorecievers you are using.

best,

Thomas

**Reply by HuijeongJeong** [reply]
Hi Thomas,

I'm so sorry, I completely forgot to follow up on this. We've disconnected and connected everything after posting this, and then weirdly the noise went away. Thanks for offering help though!

It is totally unrelated to this, but we have another question. We recently upgraded our GUI to version 1, and under 2EM_1EM_pulsed mode, the GUI doesn't seem to pick up the DIGITAL2 input from the pyPhotometry board? Instead, it saves DIGITAL1 input as both digital_1 and digital_2 in the output file. We've tried continuous mode and 2EM_2EM_pulsed mode, and in those cases both digital inputs were correctly saved. Do you have any idea what could be wrong by any chance? 

Thanks for your help in advance!!

Best,
Huijeong

**Comment by ThomasAkam** [comment]
Hi @HuijeongJeong,

Glad to hear the noise issue has been resolved.

Many thanks for reporting the bug with Digital 2 in 2EX_2EM_pulsed mode in version 1.0.1.  I have just pushed a commit and done a new release [v1.0.2](https://github.com/pyPhotometry/code/releases/tag/v1.0.2) which should fix this bug.  Please can you give it a try and let me know if the issue is resolved.

thanks,

Thomas

**Reply by HuijeongJeong** [reply]
Perfect, thanks a lot! I'll test it this afternoon and let you know how it goes!

**Reply by namboodirilab** [reply]
Hi Thomas,

I just tried and it works perfect! I appreciate it again!

Thanks, 
Huijeong
